src/Orleans.CodeGenerator/CopierGenerator.cs

```diff
@@ -223,31 +232,30 @@ static ExpressionSyntax GetService(TypeSyntax type)
                 }
             }
 
-            for (var hookIndex = 0; hookIndex < serializableTypeDescription.SerializationHooks.Count; ++hookIndex)
+            for (var hookIndex = 0; hookIndex < serializationHooks.Count; ++hookIndex)
             {
-                var hookType = serializableTypeDescription.SerializationHooks[hookIndex];
+                var hookType = serializationHooks[hookIndex];
                 fields.Add(new SerializationHookFieldDescription(hookType.ToTypeSyntax(), $"_hook{hookIndex}"));
             }
 
             return fields;
         }
 
-        public static IEnumerable<CopierFieldDescription> GetCopierFieldDescriptions(IEnumerable<IMemberDescription> members, LibraryTypes libraryTypes)
+        public static void GetCopierFieldDescriptions(IEnumerable<IMemberDescription> members, LibraryTypes libraryTypes, List<GeneratedFieldDescription> fields)
         {
-            var filteredMembers = members
-                .Distinct(MemberDescriptionTypeComparer.Default)
-                .Where(t => !libraryTypes.StaticCopiers.Any(c => SymbolEqualityComparer.Default.Equals(c.UnderlyingType, t.Type)));
-
             var fieldIndex = 0;
-            foreach (var member in filteredMembers)
-            {
-                yield return GetCopierDescription(member, fieldIndex++, libraryTypes);
-            }
-
-            static CopierFieldDescription GetCopierDescription(IMemberDescription member, int fieldIndex, LibraryTypes libraryTypes)
+            foreach (var member in members.Distinct(MemberDescriptionTypeComparer.Default))
             {
-                TypeSyntax copierType = null;
                 var t = member.Type;
+
+                if (libraryTypes.IsShallowCopyable(t))
+                    goto skip;
```

Comment:
Why not `continue`?

Reply:
No reason - I added this condition after the loop below and didn't notice that `continue` would work here.